app/models/piece.rb

```diff
@@ -73,4 +74,24 @@ def move_to!(new_x, new_y)
       self.update(x_position: new_x, y_position: new_y)
     end
   end
+=======
+  def moving_on_board?(x, y)
+    x.between?(0, 7) && y.between?(0, 7)
+  end
+
+  def valid_move?
+    if !(self.moving_on_board?(x, y))
+      return false
+    elsif self.is_obstructed?(x, y) && @pieces_in_the_way.count > 1
```

Comment:
Should this be greater then 0 or `>= 1` since we want 0 pieces in the way. Maybe this is for line 89 and occupying pieces?

Reply:
That does make sense, l was thinking about opponents but that wouldn't make sense.

Reply:
I was just looking back and I did put >1 because I was thinking of opponents and a piece (not necessarily a king but something that could go multiple spots) couldn't go through more than 1 opponent.